frontend/src/components/NavBar.js

```diff
@@ -92,6 +93,15 @@ class NavBar extends React.Component<{
 
     createNewWorksheet() {
         this.resetDialog();
+        if (!NAME_STR_REGEX.test(this.state.newWorksheetName)) {
+            this.setState({
+                snackbarShow: true,
+                snackbarMessage: `Worksheet name with name ${this.state.newWorksheetName} may not contain spaces`,
```

Comment:
We should make the message the same as on the backend: https://github.com/codalab/codalab-worksheets/blob/69ee12fa74c0699cfe9d3f998179ef85959a5eef/codalab/lib/spec_util.py#L58